src/Network/X509/SelfSigned.hs

```diff
@@ -450,9 +470,21 @@ unsafeFingerprintFromText = fromJuste . fingerprintFromText . T.pack
 -- PEM Encoded Certificate
 
 newtype X509CertPem = X509CertPem B.ByteString
-    deriving (Show, Eq, Ord, Generic)
+    deriving (Show, Generic)
     deriving anyclass (Hashable, NFData)
 
+instance Eq X509CertPem where
```

Comment:
`trim` in Eq sounds bad from a perf POV plus, why do we expect these things to not be well behaved in the first place. A smart constructor sounds like a better approach.

Reply:
We don't own the PEM encoding. The standard (RFC 7468) requires parsers to ignore white space. In this PR in only trim at the front end the end (mostly to account for the optional '\n' at the end of files), and to make the tests pass. (For the line length, 64 characters seem to be the de facto standard.)

However, I fully standard compliant implementation would strip all white space. Should I update the PR to strip all white space?

Reply:
Performance doesn't really matter here, since in operation we only store and compare the fingerprints of certificates. It's possible that this instance is only used for the encoding roundtrip tests.

Reply:
Would prefer one-time strip instead of constantly doing in `Eq`

Reply:
I updated the code to strip on construction. It's still a little fiddly. A better solution would probably always decode into DER, but that seems overkill for the use case here.

Btw. my original implementation of `trim` in this PR was completely wrong, I hope, I got it right this time :-)